views.py

```diff
@@ -1,45 +1,91 @@
 import random
 import datetime
+import logging
 
-from django.shortcuts import render_to_response
+from django.shortcuts import render
 from django.template import RequestContext
 
 from onisite.plugins.featured_content import config
 from core import models
 
 def featured(request):
-    number = config.NUMBER
-    pages = None
-    if config.THISDAY:
-        pages = _this_day()
-        if not pages is None:
-            this_day_title = True
-    if pages is None and config.RANDOM:
-        page_len = len(models.Page.objects.all())
-        select_pages = random.sample(xrange(1, page_len), number)
-        pages = map(_get_page_info_by_id, select_pages)
-    elif pages is None:
-        # grab randomly from the curated selection
-        page_info = config.PAGES
-        page_len = len(page_info)
-        if page_len <= number:
-            pages = map(_get_page_by_info, page_info)
-        else:
-            rand_nums = random.sample(range(1, page_len), number)
-            random_pages = []
-            for num in rand_nums:
-                random_pages.append(page_info[num])
-            pages = map(_get_page_by_info, random_pages)
-
-    return render_to_response('featured.html',
-                           dictionary=locals(),
-                           context_instance=RequestContext(request))   
+    # Seed the RNG with today's date so we always feature the same page(s) for
+    # an entire day
+    random.seed(datetime.date.today().strftime("%Y%m%d"))
+
+    # Get and randomize pages
+    pages = []
+    all_pages, this_day_title = _get_pages()
+    if len(all_pages) > 0:
+        rand_nums = random.sample(xrange(len(all_pages)), config.NUMBER)
```

Comment:
Agreed, but honestly that's the least of our problems.  The plugin should probably be using django.settings, too, instead of its own config.  There are other settings (here and in core) that have terrible names.  The bigger issue now is that as there are people standing this up, changing config names screws those people....

We're not *supposed* to be production-ready, but we sort of are :-/